### ISSUE TITLE
[BUG] Compilation of Protox 1.6 fails

### ISSUE THREAD

**Issue by sneako** [ahamez/protox#76]
Congrats on the new release! I was trying to update to the latest version but got the following errors. I was able to reproduce this by generating a brand new Phoenix 1.6 project and adding protox as a dependency.

**Describe the bug**
Compilation of Protox fails with the following output:

```
==> protox
Compiling 51 files (.ex)

== Compilation error in file lib/google/protobuf.ex ==
** (MatchError) no match of right hand side value: {:error, "[libprotobuf WARNING ../../../../../src/google/protobuf/compiler/parser.cc:651] No syntax specified for the proto file: 3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto. Please use 'syntax = \"proto2\";' or 'syntax = \"proto3\";' to specify a syntax version. (Defaulted to proto2 syntax.)\ngoogle/protobuf/any.proto: File not found.\ngoogle/protobuf/duration.proto: File not found.\ngoogle/protobuf/field_mask.proto: File not found.\ngoogle/protobuf/struct.proto: File not found.\ngoogle/protobuf/timestamp.proto: File not found.\ngoogle/protobuf/wrappers.proto: File not found.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:1:1: Import \"google/protobuf/any.proto\" was not found or had errors.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:2:1: Import \"google/protobuf/duration.proto\" was not found or had errors.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:3:1: Import \"google/protobuf/field_mask.proto\" was not found or had errors.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:4:1: Import \"google/protobuf/struct.proto\" was not found or had errors.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:5:1: Import \"google/protobuf/timestamp.proto\" was not found or had errors.\n3EE60D098B85EF5C3B580784A65632E262F3BC4B.proto:6:1: Import \"google/protobuf/wrappers.proto\" was not found or had errors.\n"}
    expanding macro: Protox.__using__/1
    lib/google/protobuf.ex:8: Google.Protobuf (module)
    (elixir 1.12.2) expanding macro: Kernel.use/2
    lib/google/protobuf.ex:8: Google.Protobuf (module)
could not compile dependency :protox, "mix compile" failed. You can recompile this dependency with "mix deps.compile protox", update it with "mix deps.update protox" or clean it with "mix deps.clean protox"
```

**To Reproduce**
- run `mix phx.new protox_compile_bug --no-ecto --no-html`
- open up `mix.exs`
- add `{:protox, "~> 1.6"},` to the dependencies
- `mix deps.get`
- `mix deps.compile`

**Expected behavior**
It should compile succesfully,

**Environment (please complete the following information):**
elixir          1.12.2-otp-24 
erlang          24.0.5         

I can try to take a closer look at what is going on later today if it is not immediately obvious to you what the issue could be. My hunch is that perhaps there are some new files that are not being included in the hex package?

**Comment by ahamez** [comment]
Thanks for the report! The funny thing is that I cannot reproduce this 😬. But the message returned by protoc might give a hint: `\ngoogle/protobuf/any.proto: File not found`.
I think it means that not all protoc installations come with well-known types 🤔. Out of curiosity, how did you install protoc (and on which OS) ?

I guess I have to add the definitions of these types manually. I was reluctant to do this at first, but it seems other implementations had to do this (for instance: [go](https://github.com/golang/protobuf/tree/master/ptypes)).

**Comment by ahamez** [comment]
Could you try with the  include_well_known_types branch? It should fix this bug as now well-known types proto files are embedded in protox.

**Comment by sneako** [comment]
Interesting! Maybe this has something to do with how I installed protoc? I encountered this error first on my Fedora workstation, but I just tried on my Macbook and I do not get the error there. I think I just grabbed the binary from their github releases page on both workstations...

I'll try out that branch right now!

**Comment by sneako** [comment]
The `include_well_known_types` branch works for me!

**Comment by sneako** [comment]
Just for a bit more information, in my application's dockerfile, which is based on `hexpm/elixir:1.12.3-erlang-24.0.6-ubuntu-focal-20210325` and I install protoc from `https://github.com/protocolbuffers/protobuf/releases/download/v3.15.5/protoc-3.15.5-linux-x86_64.zip`, I can reproduce the error as well when trying the released 1.6 version.

**Comment by ahamez** [comment]
Thank you! I'll release 1.6.1 right away to fix this.

BTW, looking at the readme.txt file in the archive you referenced, it says: 

> If you intend to use the included well known types then don't forget to
copy the contents of the 'include' directory somewhere as well, for example
into '/usr/local/include/'.

So I guess that some package managers don't perform this step.
